validation/src/main/scala/uk/gov/nationalarchives/tdr/validation/utils/ConfigUtils.scala

```diff
@@ -0,0 +1,166 @@
+      propertyValue.obj.get("alternateKeys") match {
+        case Some(alternateKeys) =>
+          val updatedMap = alternateKeys.arr.foldLeft(acc) { (innerAcc, key) =>
+            key.obj.foldLeft(innerAcc) { case (innerMap, (alternateKey, alternateValue)) =>
+              val existingMap = innerMap.getOrElse(alternateKey, Map.empty)
+              innerMap + (alternateKey -> (existingMap + (alternateValue.str -> propertyName)))
+            }
+          }
+          updatedMap
+        case None => acc
+      }
+    }
+    domain => key => mapped.get(domain).flatMap(_.get(key)).getOrElse(key)
+  }
+
+  /** This method takes configuration parameters and returns a function that maps a property name to an alternate key.
+    *
+    * The returned function is curried, where the first parameter is the domain, and the second parameter is the property name. It uses the configuration file to create a mapping
+    * of property names to alternate keys.
+    *
+    * @param configurationParameters
+    *   The configuration parameters containing the base schema and configuration data.
+    * @return
+    *   A curried function that takes two parameters: the domain and the property name, and returns the corresponding alternate key. Example:
+    *   `propertyToOutputMapper(configParams)("tdrFileHeader")("date_last_modified")` => `"Date last modified"`.
+    */
+  def propertyToOutputMapper(configurationParameters: ConfigParameters): String => String => String = {
+
+    val propertyToOutputsMap = configurationParameters.baseSchema("properties").obj.foldLeft(Map.empty[String, Map[String, String]]) { case (acc, (propertyName, propertyValue)) =>
+      propertyValue.obj.get("alternateKeys") match {
+        case Some(alternateKeys) =>
+          val updatedMap = alternateKeys.arr.foldLeft(acc) { (innerAcc, key) =>
+            key.obj.foldLeft(innerAcc) { case (innerMap, (alternateKey, alternateValue)) =>
+              val existingMap = innerMap.getOrElse(alternateKey, Map.empty)
+              innerMap + (alternateKey -> (existingMap + (propertyName -> alternateValue.str)))
+            }
+          }
+          updatedMap
+        case None => acc
+      }
+    }
+    domain => propertyName => propertyToOutputsMap.get(domain).flatMap(_.get(propertyName)).getOrElse(propertyName)
+  }
+
+  /** This method takes configuration parameters and returns a function that retrieves the required columns for a given domain metadata download. It uses the configuration file to
+    * create a mapping of domains to their required columns.
+    *
+    * @param configurationParameters
+    *   The configuration parameters containing the base schema and configuration data.
+    * @return
+    *   A function that takes a domain as a parameter and returns a list of tuples containing the property name and column index.
```

Comment:
Good to provide an example here as per the other methods

Reply:
added example